### PR TITLE
Remove sample tracking backend support, preserve data tables

### PR DESCRIPTION
**Pull request by guerler**
This PR removes the sample tracking features entirely from the backend. The database migration scripts remain unchanged.

### CODE REVIEW

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -5304,3 +4813,55 @@ def copy_list(lst, *args, **kwds):
         return lst
     else:
         return [el.copy(*args, **kwds) for el in lst]
+
+
+# Sample request associated class
+class Request(object, Dictifiable):
```
Comment (dannon) [inline lib/galaxy/model/__init__.py]: Out of curiosity, can we actually remove these, or does it break things if they don't exist?

Comment (guerler) [reply lib/galaxy/model/__init__.py]: These classes are used in the model mapping. Removing them without further modifications raises a missing class exception at start up.

Comment (dannon) [reply lib/galaxy/model/__init__.py]: Got an example?  Can't we remove them from the mapping, too?

**Comment by guerler** [comment]
@galaxybot test this

File: lib/galaxy/model/__init__.py (hunk 1 above)
Comment (guerler) [reply lib/galaxy/model/__init__.py]: @dannon and I talked about this. The `HistoryDatasetAssociationAnnotationAssociation` are defined in a similar fashion. In the section below, I removed the mapping and model without modifying the migration scripts. The alternative would be to leave the mapping as it is.

**Comment by jmchilton** [comment]
I'm aggressively +1 on this.

**Comment by martenson** [comment]
Thanks @guerler, good cleanup!

**Comment by guerler** [comment]
Thanks for the review guys.

**Pull request merged**